### PR TITLE
Fix bifunctor docs

### PR DESCRIPTION
**Pull request by bame-da**
Related to #7528.

- Remove non-implemented Haddock markup
- Fix code blocks
- Fix Lexer to accept `≡`
- Change Haskell REPL `>>>` to DAML REPL `daml>`

CHANGELOG_BEGIN
CHANGELOG_END

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [x] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: compiler/damlc/daml-stdlib-src/DA/Bifunctor.daml [hunk 1]
```diff
@@ -7,107 +7,107 @@ module DA.Bifunctor(Bifunctor(..)) where
...
-    -- @'bimap' f g ≡ 'first' f '.' 'second' g@
+    -- bimap f g ≡ first f . second g
     -- ````
     --
     -- Examples:
     --
-    -- ```
-    -- >>> bimap not (+1) (True, 3)
+    -- ````
+    -- daml> bimap not (+1) (True, 3)
```
Comment (cocreature) [inline compiler/damlc/daml-stdlib-src/DA/Bifunctor.daml:64]: This doesn’t trigger in `daml-doctest`.

File: compiler/damlc/daml-stdlib-src/DA/Bifunctor.daml [hunk 2]
```diff
@@ -7,107 +7,107 @@ module DA.Bifunctor(Bifunctor(..)) where
...
+-- If you supply first and second, ensure:
 --
--- ```
--- @
--- 'first' 'identity' ≡ 'identity'
--- 'second' 'identity' ≡ 'identity'
--- @
--- ```
+-- ````
+--
```
Comment (hurryabit) [inline compiler/damlc/daml-stdlib-src/DA/Bifunctor.daml:31]: Do we need an empty line here?

File: compiler/damlc/daml-stdlib-src/DA/Bifunctor.daml [hunk 3]
```diff
@@ -7,107 +7,107 @@ module DA.Bifunctor(Bifunctor(..)) where
 
 -- | A bifunctor is a type constructor that takes
 -- two type arguments and is a functor in /both/ arguments. That
--- is, unlike with 'Functor', a type constructor such as 'Either'
--- does not need to be partially applied for a 'Bifunctor'
+-- is, unlike with Functor, a type constructor such as Either
```
Comment (hurryabit) [inline compiler/damlc/daml-stdlib-src/DA/Bifunctor.daml:10]: Don't we want `Functor` to be typeset as code anymore?

Comment (bame-da) [reply compiler/damlc/daml-stdlib-src/DA/Bifunctor.daml:10]: That is not what apostrophes do in our docs. They are just rendered as text. But I'll replace them with backticks where appropriate so they become inline literals.

File: compiler/damlc/daml-stdlib-src/DA/Bifunctor.daml (hunk 1 above)
Comment (bame-da) [reply compiler/damlc/daml-stdlib-src/DA/Bifunctor.daml:64]: ok, reverting this change.

File: compiler/damlc/daml-stdlib-src/DA/Bifunctor.daml (hunk 2 above)
Comment (bame-da) [reply compiler/damlc/daml-stdlib-src/DA/Bifunctor.daml]: fixed.

**Review by cocreature — APPROVED** [review]
Thanks!

File: compiler/damlc/daml-stdlib-src/DA/Bifunctor.daml (hunk 1 above)
Comment (cocreature) [reply compiler/damlc/daml-stdlib-src/DA/Bifunctor.daml]: Opened https://github.com/digital-asset/daml/issues/7553 to improve this

File: compiler/damlc/daml-stdlib-src/DA/Bifunctor.daml (hunk 3 above)
Comment (hurryabit) [reply compiler/damlc/daml-stdlib-src/DA/Bifunctor.daml]: In Haddock proper, apostrophes typeset text as code and back it with a link to the definition of the identifier. I'm not sure we can do the latter but we should definitely do the former by using backticks. Thanks for taking care of it.

File: compiler/damlc/daml-stdlib-src/DA/Bifunctor.daml [hunk 4]
```diff
@@ -6,57 +6,56 @@ module DA.Bifunctor(Bifunctor(..)) where
...
+-- If you supply first and second, ensure:
 --
 -- ```
--- @
--- 'first' 'identity' ≡ 'identity'
--- 'second' 'identity' ≡ 'identity'
--- @
+-- first identity ≡ identity
+-- second identity ≡ identity
+--
```
Comment (richardkapolnai-da) [inline compiler/damlc/daml-stdlib-src/DA/Bifunctor.daml:33]: is this extra line intentional?

**Review by richardkapolnai-da — APPROVED** [review]

**Pull request merged**